web_app_client.py

```diff
@@ -0,0 +1,173 @@
+        """
+        Get the status of a request by UID or by email and a filename.
+        :param filename: The name of the file uploaded.
+        :param email: The email address used during the upload.
+        :param uid: The unique identifier (UID) of the request.
+        :return: Status Object representing the status of the request.
+        :raises Exception: If the status request fails.
+        """
+        try:
+            if uid:
+                upload_identified = uid
+            elif email and filename:
+                upload_identified = f"{filename} {email}"
+            else:
+                raise Exception(f"Status request failed: no upload identified provided")
+            response = requests.get(f"{self.base_url}/status/{upload_identified}")
+            response_json = response.json()
+            if response.ok:
+                status = response_json['status']
+                filename = response_json['filename']
+                timestamp = response_json['upload_time']
+                explanation = []
+                if response_json['status'] == 'done':
+                    explanation = response_json['explanation']
+                return Status(status, filename, timestamp, explanation)
+            else:
+                raise Exception(f"Status request failed")
+        except Exception as e:
+            raise Exception(f"Status request failed: {str(e)}")
+
+    def check_status_for_user(self, file_name: str, email: str) -> Status:
+        """
+        Retrieves the status of an upload from the Web API.
+
+        Args:
+            file_name (str): The file name.
+            email (str): The user email.
+        Returns:
+            Status: The status of the upload.
+        Raises:
+            requests.HTTPError: If the status request fails.
+        """
+        url = f"{self.base_url}/status/{file_name + email}"
+        response = requests.get(url)
+
+        if response.ok:
+            response_data = json.loads(response.text)
+            return self.parse_status(response_data)
+        else:
+            response.raise_for_status()
+
+    def parse_status(self, response_data: Dict[str, Union[str, int]]) -> Status:
+        """
+        Parses the status response data into a Status object.
+
+        Args:
+            response_data (Dict[str, Union[str, int]]): The response data from the Web API.
+
+        Returns:
+            Status: The parsed status object.
+        """
+        status = response_data['status']
+        filename = response_data['filename']
+        timestamp = datetime.fromtimestamp(response_data['timestamp'])
+        explanation = response_data['explanation']
+
+        return Status(status, filename, timestamp, explanation)
+
+
+def run_web_app():
+    client = WebAppClient('http://localhost:5000')
+
+    # Upload a file and get the UID
+    uid = client.upload(r"C:\Networks\final_project_test\asyncio-intro.pptx")
+    print(f"Upload UID: {uid}")
+    time.sleep(40)
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.